src/planning/src/planning/velocity_planning.cpp

```diff
@@ -0,0 +1,87 @@
+#include "planning/velocity_planning.hpp"
+
+double VelocityPlanning::find_circle_center(PathPoint &point1, PathPoint &point2,
+                                            PathPoint &point3) {
+  double x1 = point1.position.x;
+  double y1 = point1.position.y;
+  double x2 = point2.position.x;
+  double y2 = point2.position.y;
+  double x3 = point3.position.x;
+  double y3 = point3.position.y;
+
+  PathPoint mid1 = PathPoint((x1 + x2) / 2, (y1 + y2) / 2, 0);
+  PathPoint mid2 = PathPoint((x2 + x3) / 2, (y2 + y3) / 2, 0);
+  double slope1 = (x2 != x1) ? ((y2 - y1) / (x2 - x1)) : MAXFLOAT;
+  double slope2 = (x3 != x2) ? ((y3 - y2) / (x3 - x2)) : MAXFLOAT;
+  double slope1_perpendicular = -1 / slope1;
+  double slope2_perpendicular = -1 / slope2;
```

Comment:
This breaks when y1=y2 or y3=y2. Please change to something like "slope1_perpendicular = (y1!=y2) ? -1.0 / slope1 : MINFLOAT; and the same thing for slope2_perpendicular

Reply:
fixed with:
if ((slope1<=0.001) || (slope2<=0.001)) {
    return 10000;
  }

if the slope is 0, or very low, it will just return a large value because there is no need to spend resources calculating the real value

Reply:
I think this works:

double slope1 = (x2 != x1) ? (std::max((y2 - y1) / (x2 - x1), 0.0001)) : MAXFLOAT;
double slope2 = (x3 != x2) ? (std::max((y3 - y2) / (x3 - x2), 0.0001)) : MAXFLOAT;

Reply:
That would make all negative slopes be calculated as 0.0001